[inductor] Handle the case where kwargs contains tensor

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #88215

Summary: Fix https://github.com/pytorch/torchdynamo/issues/1805;
currently inductor does not allow any tensor in kwargs.

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx

desertfire committed:
[inductor] Handle the case where kwargs contains tensor

Summary: Fix https://github.com/pytorch/torchdynamo/issues/1805;
currently inductor does not allow any tensor in kwargs.

[ghstack-poisoned]

ngimel commented on torch/_inductor/lowering.py:
```diff
@@ -164,8 +164,6 @@ def wrapped(*args, **kwargs):
             args = args[0]
         # Only look at args that are Tensors
         indices = [i for i, x in enumerate(args) if isinstance(x, TensorBox)]
-        # kwargs tensors not supported yet
```
what would happen if regular (non-fallback) kernel has tensor kwargs?

desertfire replied:
Then we need to use pytree here too. Since we haven't hit the case yet, mind if I just alter the assert condition here to exclude the fallback ops, and come back to update when we hit the case later?

ngimel replied:
Yes, absolutely

desertfire committed:
Update on "[inductor] Handle the case where kwargs contains tensor"


Summary: Fix https://github.com/pytorch/torchdynamo/issues/1805;
currently inductor does not allow any tensor in kwargs.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge -f "the deploy CI failure is irrelevant"

huydhn commented:
@pytorchbot revert -m 'Sorry for reverting your PR but I think it breaks trunk https://github.com/pytorch/pytorch/actions/runs/3380662072/jobs/5613987333 with a failure in test_torchinductor_opinfo.py` -c nosignal

huydhn commented:
@pytorchbot revert -m 'Sorry for reverting your PR but I think it breaks trunk https://github.com/pytorch/pytorch/actions/runs/3380662072/jobs/5613987333 with a failure in test_torchinductor_opinfo.py' -c nosignal

desertfire commented:
@pytorchbot rebase

desertfire committed:
Update on "[inductor] Handle the case where kwargs contains tensor"


Summary: Fix https://github.com/pytorch/torchdynamo/issues/1805;
currently inductor does not allow any tensor in kwargs.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Handle the case where kwargs contains tensor"


Summary: Fix https://github.com/pytorch/torchdynamo/issues/1805;
currently inductor does not allow any tensor in kwargs.

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx

[ghstack-poisoned]
